### ISSUE TITLE
[feature request] Stop torrents after running time

### ISSUE THREAD

**Issue by nhsqr** [transmission/transmission#111]
It would be great if you can choose between "Stop seeding if idle for (min):" and "Stop seeding after running time (min):".
The "Running time" info is already in the Inspector so an additional option below "Stop seeding if idle for (min):" should not be so difficult to add probably.
Thoughts?

**Comment by xelra** [comment]
This feature has already been requested 6 years ago. There's a long running [ticket about it on trac](https://trac.transmissionbt.com/ticket/3779). It has a patch, but it's missing api endpoints and a representation in the UI.

I'm running on that patch for a long time now, but it would be really nice if the implementation could be finished and made available to regular users.
I get asked about this almost on a weekly basis. This feature really is in high demand.

**Comment by cfpp2p** [comment]
> ...but it's missing api endpoints and a representation in the UI...

Additionally, the [patch on trac](https://trac.transmissionbt.com/attachment/ticket/3779/transmission.patch) is not versatile.  It is session wide only to encompass every seeding torrent, and it doesn't take into account if torrents have been stopped for a period of time then restarted.

**Comment by xelra** [comment]
Any progress on this?

I was really excited when @mikedld said that this was easy to implement and it appeared to me that this was finally happening.

I've been waiting for this feature for a very long time, even tried to implement it myself (and failed) with 0 Objective-C knowledge. I really hope this doesn't take another 7 years.

**Comment by CPngN** [comment]
Yes pls.  So many "stalled" downloads that are complete "as far as I care about (ie: the files are intact on my side") just sitting there keeping other processes from moving on because no one is downloading bits from my side and likely never will.  I often have to go in to an individual torrent and drop the seeding % so that it will finish and close up shop as expected.

**Comment by dethmourne** [comment]
/signed. I want to say "Seed all torrents until ratio hits X or they have been seeding for Y days (regardless of activity)" and I think having a user-friendly way to implement this would be fantastic.

**Comment by theerror** [comment]
This is really sad... would help some money motivation? I'm willing to pay even 40$ for this, to have... is it really so hard?

**Comment by pataquets** [comment]
@theerror : You might find [Bountysource](https://www.bountysource.com/) useful.

**Comment by Justo-Lobo** [comment]
> /signed. I want to say "Seed all torrents until ratio hits X or they have been seeding for Y days (regardless of activity)" and I think having a user-friendly way to implement this would be fantastic.

+1 for this exact functionality.

**Comment by frankea** [comment]
Adding a +1 to this 10 year old request.  This is the only feature missing from Transmission that makes me end up using a different client.

**Comment by Waldorf3** [comment]
This, +1

Many (most?) private trackers have a minimum seed time, something like 72 hours. It would be nice to have a way to set this in transmission.

**Comment by erik1o6** [comment]
Was this ever added?

**Comment by SolidAxel** [comment]
> Was this ever added?

No, it doesn't seem like it.

**Comment by stunrelay** [comment]
Is this feature on the horizon? Absolute deal breaker if using PTs.

**Comment by erik1o6** [comment]
Anyone want to work on it together?

**Comment by parer55** [comment]
Truly a must have. Would love to see it coming to transmission. Thanks to anyone who can make it happen.

**Comment by sameep22** [comment]
Just wanted to add my support for this feature... This would make transmission the perfect client for my set up.

**Comment by killemov** [comment]
> This feature has already been requested 6 years ago. There's a long running [ticket about it on trac](https://trac.transmissionbt.com/ticket/3779). It has a patch, but it's missing api endpoints and a representation in the UI.
> 
> I'm running on that patch for a long time now, but it would be really nice if the implementation could be finished and made available to regular users. I get asked about this almost on a weekly basis. This feature really is in high demand.

I did not reference [this forum topic](https://forum.transmissionbt.com/viewtopic.php?t=10364) then, but it is still somewhat scriptable NOW as demonstrated in the "Unrar and cleanup script".

**Comment by rak00001** [comment]
Any developments on this at all as far as anybody knows?

**Comment by Softchair** [comment]
Any update on this? Would really love to see it implemented. I could throw in some time if needed, but my expertise isn’t applications
